dataregistry-extension/src/active.ts

```diff
@@ -42,6 +42,18 @@ export default {
   autoStart: true
 } as JupyterFrontEndPlugin<IActiveDataset>;
 
+/**
+ * If the "active" URL is an empty string or the active url
+ * itself, replace it will null. This prevens infinite recursion
+ * of the active URL and  trying to parse empty URLs.
+ */
+function processActive(url: string | null): string | null {
```

Comment:
Can the input argument be `null`?

If not, I would suggest that this function take and return strings; otherwise, this function is liable to be de-optimized by runtime compilers due to changing input and output value types.

Based on the ternary condition in L77, whether the `url` is `null` or an empty string, the result should be the same, namely an empty `Set`.